hax/test/integration/test_motr.py

```diff
@@ -177,3 +242,208 @@ def my_services(name):
+                ('m0conf/nodes/0x6e00000000000001:0x3/processes'
+                 '/0x7200000000000001:0x15/services/0x7300000000000001'
+                 ':0x17',
+                 "{\"name\": \"ios\", \"state\": \"failed\"}"),
+                ('m0conf/nodes/0x6e00000000000001:0x3/processes'
+                 '/0x7200000000000001:0xa/services/0x7300000000000001:0xc',
+                 "{\"name\": \"ios\", \"state\": \"failed\"}")]
+            ]
+        elif (key == 'm0conf/nodes/0x6e00000000000001:0x3/processes'
+              '/0x7200000000000001:0x15' and recurse):
+            return [new_kv(k, v) for k, v in [(
+                'm0conf/nodes/0x6e00000000000001:0x3/processes'
+                '/0x7200000000000001:0x15',
+                "{\"name\": \"m0_server\", \"state\": \"failed\"}"),
+                ('m0conf/nodes/0x6e00000000000001:0x3/processes'
+                 '/0x7200000000000001:0x15/services/0x7300000000000001'
+                 ':0x17',
+                 "{\"name\": \"ios\", \"state\": \"failed\"}")]
+            ]
+        elif (key == 'm0conf/nodes/0x6e00000000000001:0x3/processes'
+              '/0x7200000000000001:0x15'):
+            return new_kv('m0conf/nodes/0x6e00000000000001:0x3/processes'
+                          '/0x7200000000000001:0x15',
+                          '{\"name\": \"m0_server\", \"state\": \"failed\"}')
+        elif (key == 'm0conf/nodes/0x6e00000000000001:0x3/processes'
+              '/0x7200000000000001:0xa'):
+            return new_kv('m0conf/nodes/0x6e00000000000001:0x3/processes'
+                          '/0x7200000000000001:0xa',
+                          '{\"name\": \"m0_server\", \"state\": \"offline\"}')
+        elif key == 'm0conf/nodes/localhost/processes/7/services/rms':
+            return new_kv('m0conf/nodes/localhost/processes/7/services/rms',
+                          '17')
+        elif key == 'm0conf/nodes/0x6e00000000000001:0x3':
+            return new_kv('m0conf/nodes/0x6e00000000000001:0x3',
+                          '{\"name\": \"localhost\", '
+                          '\"state\": \"M0_NC_UNKNOWN\"}')
+        raise RuntimeError(f'Unexpected call: key={key}, recurse={recurse}')
+
+    mocker.patch.object(consul_util.kv, 'kv_get', side_effect=my_get)
+    # TODO: Handle 'kv_put' by updating kv returned by 'kv_get'
+    mocker.patch.object(consul_util.kv, 'kv_put', return_value=0)
+    mocker.patch.object(consul_util,
+                        'get_node_fid',
+                        return_value=Fid(0x6e00000000000001, 0x3))
+    mocker.patch.object(consul_util,
+                        'get_node_encl_fid',
+                        return_value=Fid(0x6500000000000001, 0x4))
+
+    motr.broadcast_ha_states([HAState(fid=Fid(0x7200000000000001, 0x15),
+                                      status=ServiceHealth.FAILED)])
+
+    traces = motr._ffi.traces
+    assert AssertionPlan(tr_and(tr_method('ha_broadcast'),
```

Comment:
I would also like to see that in such a case NO node failure notification happened. In other words, there is no node fid in the broadcast (is it a correct understanding of your feature, BTW?)

Reply:
Done